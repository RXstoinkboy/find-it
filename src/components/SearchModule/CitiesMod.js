import React from 'react';
import styles from './styles/ResultsMod.module.scss'
import PropTypes from 'prop-types'

export const CitiesMod = props => {
    return (
        // display the box only if input is focused
        <div className={props.focus && props.city.length > 0 ? [styles.wrapper, styles.wrapperActive].join(' ') : styles.wrapper} >
            <section className={styles.section} >
            {/* list over cites */}
                <ul className={styles.contentList}>
                    {props.cities.map((city, idx) => (
                        <li 
                            onMouseDown={props.handlePickCity}
                            className={styles.contentItem} 
                            key={idx}>{city.name}</li>
                    ))}
                </ul>
            </section>
        </div>
    );
};

export default CitiesMod;

CitiesMod.propTypes = {
    city: PropTypes.string.isRequired,
    focus: PropTypes.bool.isRequired,
    handlePickCity: PropTypes.func
}