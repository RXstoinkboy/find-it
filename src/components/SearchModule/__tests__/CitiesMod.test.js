import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {CitiesMod} from '../CitiesMod'

it(`CitiesMod renders correctly`, ()=> {
    const props = {
        city: 'krakow',
        focus: false,
        cities: [1,2],
        handlePickCity: jest.fn() 
    }

    const citiesMod = shallow(
        <CitiesMod {...props} />
    )

    expect(toJson(citiesMod)).toMatchSnapshot()
})