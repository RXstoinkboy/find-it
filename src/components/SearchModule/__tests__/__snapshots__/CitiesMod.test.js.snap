// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesMod renders correctly 1`] = `
<div
  className="wrapper"
>
  <section
    className="section"
  >
    <ul
      className="contentList"
    >
      <li
        className="contentItem"
        key="0"
        onMouseDown={[MockFunction]}
      />
      <li
        className="contentItem"
        key="1"
        onMouseDown={[MockFunction]}
      />
    </ul>
  </section>
</div>
`;
